### ISSUE TITLE
Clippy pedantic for f64 as u64

### ISSUE THREAD

**Issue by 1Dragoon** [rust-lang/rust-clippy#8601]
### Description

I just created a blank issue because I'm not quite sure what this would fall under, as it may or may not be desirable to have, so this may be more of a discussion.

Given clippy pedantic warns about losing the decimal value, could we have this warning suppressed by adding .trunc() to the f64?

Say for example:

```rust
let foo = 3.14159_f64;
let bar = foo.trunc() as u64;
```

The trunc() in this case is a no-op (I think, somebody correct me if I'm wrong) as the u64 already implicitly does that. Meanwhile we get to sidestep that particular lint without having to stick a macro there. I'd like to not suppress that lint globally in case I inadvertently try to convert a value that is a float to a type that I didn't intend and have clippy gently remind me of what I'm doing.

Some might argue though that the pedanticness of clippy should warn you that the .trunc() isn't necessary in this case. Should it be? I have no idea :)

BTW: Same thing for the sign of the value and loss of precision.

### Version

```text
rustc 1.53.0 (53cb7b09b 2021-06-17)
binary: rustc
commit-hash: 53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b
commit-date: 2021-06-17
host: x86_64-pc-windows-msvc
release: 1.53.0
LLVM version: 12.0.1
```

**Comment by mikerite** [comment]
I assume you're talking about `clippy::cast_possible_truncation`. It's not warning about the loss of decimals. It's warning about 'large values being truncated'  - [docs](https://rust-lang.github.io/rust-clippy/master/index.html#cast_possible_truncation).

```
#![warn(clippy::pedantic)]

fn main() {
    let f: f64 = 1E50;
    
    println!("{}", f);
    println!("{}", f.trunc());
    println!("{}", f.trunc() as u64);
}
```
Gives 
```
100000000000000000000000000000000000000000000000000
100000000000000000000000000000000000000000000000000
18446744073709551615
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=261a93a39a7c6278677b45269458bfc5)

**Comment by repi** [comment]
`f64` can indeed store a larger value than `u64`, so an `as` cast can be lossy and truncate high values, as @mikerite showed.

So think the lint is working correctly in this case and that this issue can be closed
